### DISCUSSION TITLE
✅ Keep swiper.min.js in lib/swiper

### DISCUSSION THREAD

**Discussion by m1ga** [elementor/elementor#29612] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

Now that v8 of Swiper is merged the `/lib/swiper/swiper.min.js` is gone and you have to use `/lib/swiper/v8/swiper.min.js` now. Looking forward when you want to update Swiper again (latest version is 11) we always have to update the path. Wouldn't it make more sense to keep the current swiper version in the `/lib/swiper/swiper.min.js` folder? That way we at least have one version we can always work with.

### Describe the solution you'd like.

Keep the current swiper version in `/lib/swiper/swiper.min.js`

### Describe alternatives you've considered.

-

### Additional context

For future Swiper updates it would make it easier for plugin devs to have the current version in `/lib/swiper/swiper.min.js` so we don't have to update the folder names all the time

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by rami-elementor** [comment]
@m1ga Why does the path matter? We are only interested in style handler name, not the file location.

When Elementor registers the style, you no longer need the path, you use only the handler name in `get_style_depends()` or in control dependencies.

**Reply by m1ga** [reply]
when I want to use Swiper in a custom widget and I want to require it I'll need to use the path. And that means I have to update it with every version if you update it to v9, v10, v11 in the future. https://github.com/elementor/elementor/issues/29616 is an issue about it

**Reply by rami-elementor** [reply]
If you want to use Swiper in your widget, you simply need to declare that the widget requires `swiper` and `e-swiper` as a dependency:

```php
class Elementor_Test_Widget extends \Elementor\Widget_Base {

	public function get_style_depends(): array {
		return [ 'swiper', 'e-swiper' ];
	}

}
```

The first style is Swiper styles and the second for Elementor's adjustments for swiper.

**Reply by m1ga** [reply]
I'll need the JS files.  If I do that and look at the source code I don't see any swiper.js file, it works fine for the styles in `get_style_depends()` but not for `get_script_depends()`

**Reply by rami-elementor** [reply]
@m1ga We investigated this issue internally, and you are correct. Elementor didn't register `swiper.js`, therefore you couldn't use it as a widget script dependency.

We decided not to change the folder structure as it may create other issue. In addition the current folder structure will help us transition to newer Swiper versions.

To address this issue, a new patch will be released [registering swiper.js script](https://github.com/elementor/elementor/pull/29656).

**Reply by m1ga** [reply]
Thanks, that helps a lot :+1: Then we don't need to have a file in the other folder when Elementor itself now handles the path for use .

**Reply by rami-elementor** [reply]
@m1ga Elementor 3.26.3 was just released, now you can use swiper.js script.

**Comment by mkabi** [comment]
Another problem has occurred in the new Elementor update.
In Elementor, I created an image swiper in the templates section, so when I want to use this image carousel on different pages, I don't have to design it every time, and I call the name of that template through the elementor templates widget to display it.
After the recent Elementor update, when this template is called through the template widget, unfortunately the images are not displayed and the corresponding script is not loaded.

![swiper](https://github.com/user-attachments/assets/84ea3624-82f2-4a1a-aab0-6f2216b9ab35)

**Comment by nicholaszein** [comment]
Hi @m1ga!

Thank you for submitting this request!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Core v3.26.3! 🥳 

**We reverted back the removal of the registration of the swiper.js library.**

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
